### DISCUSSION TITLE
Pocketbase executable takes 15 minutes to build (golang docker compose)

### DISCUSSION THREAD

**Discussion by noob8boi** [pocketbase/pocketbase#5209] [Q&A]
![image](https://github.com/pocketbase/pocketbase/assets/91360152/b6f117d2-ce37-447b-87a9-665daefd350f)

Its the step 6 where it takes a lot of time. Should I prebuild the executable and mount it? but then I won't need to do any other step other than the last one right?

**Comment by ganigeorgiev** [comment]
Based on the above screenshot it seems that you are doing a "clean" build every time when there are changes in the copied files because "Step 5" will invalidate any instructions cached after that.

A workaround could be to try to mount as volume the container's Go build caches (`GOCACHE` and `GOMODCACHE`). For example, before "Step 5" you can specify an explicit path in your image with `go env -w GOCACHE="/pb/gocache"` (_or use the default one for your container which could be found when running `go env GOCACHE` inside the container_) and then add a volume entry like `/some/local/path:/pb/gocache`). `GOMODCACHE` could also have to be defined in a similar manner.

You can also run `go help cache` to learn more about the available build caching options.

In any case, the first/non-cached build is expected to be slow due to the SQLite drivers and I don't think there is anything we can do in PocketBase to speed up the build times (suggestions are welcomed).

**Reply by noob8boi** [reply]
```
FROM golang:1.22.4

WORKDIR /pb

RUN go env -w GOCACHE=/go-cache

RUN go env -w GOMODCACHE=/gomod-cache

COPY go.mod go.sum ./

RUN --mount=type=cache,target=/gomod-cache

RUN go mod download

COPY *.go ./

RUN --mount=type=cache,target=/gomod-cache --mount=type=cache,target=/go-cache

RUN CGO_ENABLED=0 GOOS=linux go build -o /pocketbase

CMD ["/pocketbase", "serve", "--http=0.0.0.0:8090"]
```

I edited the dockerfile to this. Tried mounting directly into root directory. Is this bad?
